test/battle/move.c

```diff
@@ -82,55 +82,105 @@ SINGLE_BATTLE_TEST("Turn order is determined randomly if priority and Speed tie
+        hpMon2 = opponentLeft->hp;
+        switch (hpMon1*1000+hpMon2)
+        {
+            case 188360:
+                gBattleTestRunnerState->occurenceCounter += 1;
+                break;
+            case 189360:
+                gBattleTestRunnerState->occurenceCounter += 2;
+                break;
+            case 261360:
+                gBattleTestRunnerState->occurenceCounter += 4;
+                break;
+            case 235360:
+                gBattleTestRunnerState->occurenceCounter += 8;
+                break;
+            case 262360:
+                gBattleTestRunnerState->occurenceCounter += 16;
+                break;
+            case 202360:
+                gBattleTestRunnerState->occurenceCounter += 32;
+                break;
+            case 189378:
+                gBattleTestRunnerState->occurenceCounter += 64;
+                break;
+            case 189189:
+                gBattleTestRunnerState->occurenceCounter += 128;
+                break;
+            case 189480:
+                gBattleTestRunnerState->occurenceCounter += 256;
+                break;
+            case 188480:
+                gBattleTestRunnerState->occurenceCounter += 512;
+                break;
+            case 188240:
+                gBattleTestRunnerState->occurenceCounter += 1024;
+                break;
+            case 188188:
+                gBattleTestRunnerState->occurenceCounter += 2048;
+                break;
+            case 262262:
+                gBattleTestRunnerState->occurenceCounter += 4096;
+                break;
+            case 262142:
+                gBattleTestRunnerState->occurenceCounter += 8192;
+                break;
+            case 202403:
+                gBattleTestRunnerState->occurenceCounter += 16384;
+                break;
+            case 202202:
+                gBattleTestRunnerState->occurenceCounter += 32768;
+                break;
+            case 262283:
+                gBattleTestRunnerState->occurenceCounter += 65536;
+                break;
+            case 202283:
+                gBattleTestRunnerState->occurenceCounter += 131072;
+                break;
+            case 235180:
+                gBattleTestRunnerState->occurenceCounter += 262144;
+                break;
+            case 261180:
+                gBattleTestRunnerState->occurenceCounter += 524288;
+                break;
+            case 235235:
+                gBattleTestRunnerState->occurenceCounter += 1048576;
+                break;
+            case 235300:
+                gBattleTestRunnerState->occurenceCounter += 2097152;
+                break;
+            case 261141:
+                gBattleTestRunnerState->occurenceCounter += 4194304;
+                break;
+            case 261261:
+                gBattleTestRunnerState->occurenceCounter += 8388608;
+                break;
+        }
+        EXPECT_EQ(gBattleTestRunnerState->occurenceCounter, 16777215);
```

Comment:
I understand what's happening here: the numbers in the `case`s are the first 24 powers of 2, and we're using `+` so that if there are duplicates we'll carry, and then we're comparing to a number with all those 24 powers of 2 set.

I wonder if it might be helpful to leave a comment explaining this for future readers? I'm not sure if it's necessary to explain the how in detail, or just to explain that this is detecting whether all 24 orderings occur. Maybe it would be a little clearer with `1 << 0`, `1 << 1`, ..., and then `0xFFFFFF`? Or maybe not, I'll leave that up to you :)

Reply:
`1 << 0` `1 << 1` etc. should be compiled down to constants, right? So doing it like that doesn't change anything except  making what's happening more readable.

Reply:
Yeah, even agbcc can do [constant folding](https://en.wikipedia.org/wiki/Constant_folding) :)